frame/staking/src/lib.rs

```diff
@@ -1835,3 +2309,56 @@ impl<T, Reporter, Offender, R, O> ReportOffence<Reporter, Offender, O>
 		}
 	}
 }
+
+#[allow(deprecated)]
+impl<T: Trait> frame_support::unsigned::ValidateUnsigned for Module<T> {
+	type Call = Call<T>;
+	fn validate_unsigned(call: &Self::Call) -> TransactionValidity {
+		if let Call::submit_election_solution_unsigned(
+			winners,
+			compact,
+			validator_index,
+			signature,
+		) = call {
+			if let Some(queued_elected) = Self::queued_elected() {
+				if queued_elected.compute == ElectionCompute::Authority {
+					debug::native::debug!(
+						target: "staking",
+						"rejecting unsigned transaction because we already have one from an authority"
+					);
+					return InvalidTransaction::Future.into();
+				}
+			}
+
+			// check signature
+			let payload = (winners, compact, validator_index);
+			let current_validators = T::SessionInterface::keys::<T::KeyType>();
+			let (_, validator_key) = current_validators.get(*validator_index as usize)
+				.ok_or(TransactionValidityError::Invalid(InvalidTransaction::Custom(0u8).into()))?;
+
+			// unwrap the key if it exists.
+			let validator_key = validator_key.as_ref()
+				.ok_or(TransactionValidityError::Invalid(InvalidTransaction::Custom(1u8).into()))?;
+
+			let signature_valid = payload.using_encoded(|encoded_payload| {
+				validator_key.verify(&encoded_payload, &signature)
+			});
+
+			if !signature_valid {
+				return InvalidTransaction::BadProof.into();
+			}
+
+			Ok(ValidTransaction {
+				priority: TransactionPriority::max_value(),
+				requires: vec![],
+				// TODO: what is a good value for this?
+				provides: vec![(Self::current_era(), validator_key).encode()],
```

Comment:
This might include a hash of the solution as well - it would prevent multiple solutions to float around in the network, but is only relevant if we don't really care about the party that produced that solution.
```suggestion
				provides: vec![
				  (Self::current_era(), validator_key).encode(),
				  hash(compact),
				],
```

By having two `provides` tag you make sure that:
1. A validator can send only one solution per era
2. Same solutions are deduplicated on the transaction pool level.

Reply:
Another note. We need some replay-protection for incorrect solutions to prevent the same faulty solution to be accepted multiple times.

Reply:
@kianenigma is this addressed? ^^^

Reply:
Yes and No. We talked about it in detail offline and there are numerous ways to address this, if we are sure it is needed. Paid transaction, IMO, might not need to be deduplicated at the pool layer and they are being paid. 

I have an entry on the follow ups of the PR to generally tackle the potential issue of duplicate solutions.